// gitbook-plugin-mp3-player/index.js
var audioRegex = /^\s*```\s{0,4}mp3((.*[\r\n]+)+?)?\s*```$/im;
var audioJsStart = "<script>document.addEventListener('load', function(){playMp3List(";
var audioJsEnd = ");});</script>";
var audioTpl = '<div class="bo-wrapper"><div class="bo"><button type="button" class="bo2"><div class="bo1"><div id="player"><div class="cover"></div><div class="ctrl"><div class="tag"> <strong>Title</strong> <span class="artist">Artist</span> <span class="album">Album</span> </div><div class="control"><div class=""><div class="rewind icon"></div><div class="playback icon"></div><div class="fastforward icon"></div></div><div class="volume"><div class="mute icon left">\</div\><div class="slider left"><div class="pace"></div></div></div></div><div class="progress"><div class="slider"><div class="loaded"></div><div class="pace"></div></div><div class="timer left">0:00\</div><div class="right"><div class="repeat icon"></div><div class="shuffle icon"></div></div></div></div></div><ul id="playlist" class="uicss-cn"><span style="width: 100%;height: 7px;display: block;"></span></ul></div></button></div></div>';
var tag = 'audio-list--------';
function processAudioBlockList(page) {
  var match;

  while ((match = audioRegex.exec(page.content))) {
    var rawBlock = match[0];
    var audioContent = match[1];
    var rep = audioJsStart + audioContent + audioJsEnd + "\n";
 
    page.content = page.content.replace(rawBlock, '<div id="audio-list--------" style="display:none">\n{% raw %}\n' + audioContent + '{% endraw %}</div>');
  }

  return page;
}

module.exports = {
  book: {
        assets: './assets',
        js: [
            'jquery-3.2.1.min.js',
            'jquery-ui.min.js',
            'audio.js'
        ],
        css: [
            'audio.css'
        ]
  },
  hooks: {
    'page:before': processAudioBlockList,
  }
};
